### ISSUE TITLE
How to l2-normalize the embedding feature in flax.linen.Embed

### ISSUE THREAD

**Issue by longbowzhang** [google/flax#1729]
Hi, 
I am quite new to Jax/Flax, and I am looking forward to an example about flax.linen.Embed

- I would like to know how to normalize the embedding features with unit-norms when init the module?
- how to renormalize after each update during training? 

Thanks a lot in advance!

**Comment by matthias-wright** [comment]
Hi @longbowzhang,

Here is one way of achieving this:

Suppose you have the following simple network:

```python
class Net(nn.Module):
    
    @nn.compact
    def __call__(self, c):
        x = nn.Embed(num_embeddings=10, features=5)(c)
        x = nn.Dense(features=5)(x)
        return x
```
Then you could apply the following normalization after each training step (either in your training loop or directly in your update function):
```python
params = state.params.unfreeze()
params['params']['Embed_0']['embedding'] /= (jnp.linalg.norm(params['params']['Embed_0']['embedding'], axis=-1, keepdims=True) + 1e-10)
state = state.replace(params=frozen_dict.freeze(params))
```
Of course you can apply the same normalization after initializing your parameters.

Let me know if you need more details, I could create a small Colab for you.

Hope that helps!
